backend/app/usecase/presenter_mock.go

```diff
@@ -0,0 +1,58 @@
+package usecase
+
+type showHomeCallArgs struct {
+	authToken string
+}
+
+type showUserHomeCallArgs struct {
+	authToken string
+}
+
+type showExternalPageCallArgs struct {
+	link string
+}
+
+var _ Presenter = (*mockPresenter)(nil)
+
+type mockPresenter struct {
+	showHomeCallArgs         []showHomeCallArgs
+	showUserHomeCallArgs     []showUserHomeCallArgs
+	showExternalPageCallArgs []showExternalPageCallArgs
+}
+
+func (m *mockPresenter) ShowHome() {
+	m.showHomeCallArgs = append(m.showHomeCallArgs, showHomeCallArgs{})
+}
+
+func (m *mockPresenter) ShowUserHome(authToken string) {
+	m.showUserHomeCallArgs = append(
+		m.showUserHomeCallArgs,
+		showUserHomeCallArgs{authToken: authToken},
+	)
+}
+
+func (m mockPresenter) ShowAliasNotFound() {
+	panic("implement me")
+}
+
+func (m mockPresenter) ShowLongLink(longLink string) {
+	panic("implement me")
+}
+
+func (m *mockPresenter) ShowExternalPage(link string) {
+	m.showExternalPageCallArgs = append(
+		m.showExternalPageCallArgs,
+		showExternalPageCallArgs{link: link},
+	)
+}
+
+func (m mockPresenter) ShowInvalidAuthTokenError() {
+	panic("implement me")
+}
+
+func newMockPresenter() mockPresenter {
+	return mockPresenter{
+		showUserHomeCallArgs:     make([]showUserHomeCallArgs, 0),
+		showExternalPageCallArgs: make([]showExternalPageCallArgs, 0),
```

Comment:
Here I see 2 problems:
* `showHomeCallArgs` is not initialized
* we should negotiate how to initialize empty slice:
```
1. make([]showUserHomeCallArgs, 0)
2. []showUserHomeCallArgs{}
```

Reply:
@alldroll I saw most of the docs uses `make` to init empty struct. What are your thoughts?

Reply:
Done.

Reply:
I guess there is no difference. We just have to decide what to use ^)